src/chat.rs

```diff
@@ -3069,10 +3073,12 @@ pub(crate) async fn create_send_msg_jobs(context: &Context, msg: &mut Message) -
         }
     }
 
-    if rendered_msg.is_encrypted && !needs_encryption {
+    if rendered_msg.is_encrypted {
         msg.param.set_int(Param::GuaranteeE2ee, 1);
-        msg.update_param(context).await?;
+    } else {
+        msg.param.remove(Param::GuaranteeE2ee);
     }
+    msg.update_param(context).await?;
```

Comment:
This `update_param()` isn't always necessary. But i'd suggest to just merge it with `update_subject()` below to make the only transaction